presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -150,6 +154,20 @@ public SqlFunctionProperties getSqlFunctionProperties()
         return sessionPropertyManager.decodeCatalogPropertyValue(connectorId, catalog, propertyName, properties.get(propertyName), type);
     }
 
+    /**
+     * Comma separated headers
+     * @return List of headers or Optional.empty if no header is injected
+     */
+    @Override
+    public Optional<List<String>> getMetastoreHeaders()
+    {
+        String metaStoreHeaders = session.getConnectorProperties(connectorId).get(METASTORE_HEADERS);
+        if (metaStoreHeaders == null) {
+            return Optional.empty();
+        }
+        return Optional.of(METASTORE_HEADER_SPLITTER.splitToList(metaStoreHeaders));
```

Comment:
This can be written as:

```
        Optional<String> metaStoreHeaders = Optional.ofNullable(session.getConnectorProperties(connectorId).get(METASTORE_HEADERS));
        return metaStoreHeaders.map(METASTORE_HEADER_SPLITTER::splitToList);
```

Reply:
Actually why not add a new session property in `HiveSessionProperties` ?

Reply:
✅